boot/zephyr/Kconfig

```diff
@@ -495,6 +495,10 @@ config MCUBOOT_HW_DOWNGRADE_PREVENTION
 
 endchoice
 
+config BOOT_WATCHDOG_FEED
+	bool "Feed the watchdog while doing swap"
+	depends on NRFX_WDT
```

Comment:
Why should this depend on a specific vendor implementation of the watchdog?

If this is needed it should probably default to `y` to match current behavior.

Reply:
yes, and no

Indeed implementation allows what you suggested but for other platforms the feed is not implemented.
I think better to not suggest to user that something is implemented while is not.

Reply:
It's not implemented upstream.  You have no idea what patches are applied downstream.

Reply:
A good reason to have this option is in a case where it is implemented, but is not desired.  For that it should not have a dependency on a vendor, and should default to enabled.

Reply:
Alternatively, leave it defaulted to disabled and change the code so that if it's selected and there isn't a vendor-specific definition of `MCUBOOT_WATCHDOG_FEED` it hits an `#error` or other build-time assert.  Only when it's not selected should the existing no-op empty definition be used.

Is that more reasonable?

Reply:
I would set this to `y` as you initially suggested.

Reply:
Yes, that's the safe approach to match existing behavior.  Sorry, I'm jumping between too many things and not paying enough attention to subtleties.